flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -641,4 +610,32 @@ static void require(boolean condition, String message, Object... values) {
 		}
 	}
 
+	/**
+	 * Get dynamic properties based on two Flink configuration. If base config does not contain and target config
+	 * contains the key or the value is different, it should be added to results. Otherwise, if the base config contains
+	 * and target config does not contain the key, it will be ignored.
+	 * @param baseConfig The base configuration.
+	 * @param targetConfig The target configuration.
+	 * @return Dynamic properties as string, separated by space.
+	 */
+	static String getDynamicProperties(
```

Comment:
Q: do these stringify arguments stored somewhere during the container launch? if yes, are they stored/managed by the NM thread? 

If that's a yes, say users configure a rather large `flink-conf.yaml`, would this be a problem at all? (consider we have one stringify representation per container)

Reply:
No, it does not store anywhere. It will be passed to the launch command of `TaskExecutor`. We only get the differences with client uploaded flink-conf.yaml, and then pass it to `TaskExecutor` by dynamic properties.